### ISSUE TITLE
picoquic refuses to reply to zero-length connection IDs

### ISSUE THREAD

**Issue by marten-seemann** [private-octopus/picoquic#480]
When using a zero-length CID as the SCID on the Initial packet, the picoquic server uses the DCID from that Initial as the DCID to send the response to. It should instead respect the client's choice to use a zero-length CID.

Example log: https://test.privateoctopus.com/cgi-bin/test.pl?name=fe897142cd912a13a5118ccdb539

**Comment by huitema** [comment]
@marten-seemann the fix is being tested in PR #482. As usual, the root cause is "if it wasn't tested, it probably does not work". So I added a test that tries setting connections of various CID lengths. I also tightened the verification of DCID in the incoming packets, which reproduced your issue. And then I found a couple of bugs that had to be fixed.

The good part is that the picoquic clients can now be configured to use any valid connection ID length -- but I have not yet added that argument to the demo client.
